### DISCUSSION TITLE
How to log pruning sparsity in wandb logger

### DISCUSSION THREAD

**Discussion by whtitefall** [Lightning-AI/pytorch-lightning#11104] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,

I'm trying to use model pruning in lightning, I know the verbose parameter log out the sparsity of the model layers, 

but how do I log the sparsity versus epoch/steps into the logger? is there a customized logger function that I need to specify? 

```
wandb_logger = pl.loggers.wandb.WandbLogger()

trainer = pl.Trainer.from_argparse_args(args, 
    logger=wandb_logger,
    callbacks=[ModelPruning("l1_unstructured", amount=0.2,verbose=2,)]
    )
trainer.fit(model, dm)

```

**Comment by rohitgr7** [comment]
hey!
logging pruning stats isn't supported natively since it's an extension,, but you can use `self.logger.experiment.log` to log any kind of stats supported by wandb. To access the step you can use `self.global_step/self.current_epoch`.
